src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -167,20 +109,69 @@ public void testDesktopSharingInPresence()
         catch (TimeoutException e)
         {
             assertEquals(
-                expectedResult, remoteVideoType[0],
-                "Wrong video type, maybe desktop sharing didn't work");
+                    expectedResult, remoteVideoType[0],
+                    "Wrong video type, maybe desktop sharing didn't work");
         }
+    }
 
-        // allow the participant to leave
-        try
-        {
-            waitEndSignal.await(10, TimeUnit.SECONDS);
-            print("End DesktopSharingTest.");
-        }
-        catch (InterruptedException e)
-        {
-            e.printStackTrace();
-        }
-        ensureTwoParticipants();
+    /**
+     * Starts desktop sharing.
+     */
+    private void startDesktopSharing()
+    {
+        TestUtils.waitForElementByXPath(
+                getParticipant2().getDriver(),
+                DS_BUTTON_XPATH,
+                5);
+        MeetUIUtils.clickOnToolbarButton(getParticipant2().getDriver(),
+                DS_BUTTON_ID);
+        TestUtils.waitForElementContainsClassByXPath(
+                getParticipant2().getDriver(), DS_BUTTON_XPATH,
+                "toggled", 2);
+    }
+
+    /**
+     * Stops desktop sharing.
+     */
+    private void stopDesktopSharing()
+    {
+        TestUtils.waitForElementByXPath(
+                getParticipant2().getDriver(),
+                DS_BUTTON_XPATH,
+                5);
+        MeetUIUtils.clickOnToolbarButton(getParticipant2().getDriver(),
+                DS_BUTTON_ID);
+        TestUtils.waitForElementNotContainsClassByXPath(
+                getParticipant2().getDriver(), DS_BUTTON_XPATH,
+                "toggled", 2);
+    }
+
+    /**
+     * Checks the video layout on the other side, after we imitate
+     * desktop sharing.
+     * @param isScreenSharing <tt>true</tt> if SS is started and <tt>false</tt>
+     *                        otherwise.
+     */
+    private void checkExpandingDesktopSharingLargeVideo(boolean isScreenSharing)
+    {
+        // check layout
+        new VideoLayoutTest().driverVideoLayoutTest(
+            getParticipant1(), isScreenSharing);
+
+        // hide thumbs
+        MeetUIUtils.clickOnToolbarButton(
+            getParticipant1().getDriver(), "toggleFilmstripButton");
+
+        TestUtils.waitMillis(5000);
```

Comment:
What are we waiting here for exactly ? Waits like this are waste of time and never check for the condition that is expected after the wait period is over.

Reply:
We will leave this for now like this. Hiding the filmstrip because of animations can take some time. We cannot check for a css change, cause currently, we change the css and after the change, it starts moving the filmstrip away from the screen for two seconds (which with the framebuffer can take more). We already spend a lot of time looking at it to figure it out. Maybe checking for isDisplayed of the local video is one solution will give it a quick try and if it doesn't work will leave the waits.

Reply:
So the purpose could be explained in a comment at least.

Reply:
Seems to wait for local video not displayed works. Will see how it goes. Will merge it after the tests pass.